src/main/java/org/elasticsearch/cluster/metadata/MetaDataMappingService.java

```diff
@@ -386,9 +386,20 @@ public ClusterState execute(final ClusterState currentState) throws Exception {
                                 if (mergeResult.hasConflicts()) {
                                     throw new MergeMappingException(mergeResult.buildConflicts());
                                 }
+                            } else {
+                                // TODO: can we find a better place for this validation?
```

Comment:
Ok, I see the problem. Yes, I think really all the stuff happening in the metadata index service for mappings should be moved into the mapper. So the if multiple types are being added, whether for index creation or mappings update, validation can happen within the mapper service. But I can handle this as a separate issue later.